### PR TITLE
PCVL-798 Feed-forward configurator

### PR DESCRIPTION
**Pull request by ericbrts**

### CODE REVIEW

File: perceval/components/abstract_processor.py [hunk 1]
```diff
@@ -252,13 +254,43 @@ def add(self, mode_mapping, component, keep_port: bool = True) -> AProcessor:
...
+        if min(photonic_modes) < 0 or max(photonic_modes) >= self.m:
+            raise ValueError(f"Mode numbers must be in [0; {self.m - 1}] (got {photonic_modes})")
+        if any([self._mode_type[i] != ModeType.PHOTONIC for i in photonic_modes]):
+            raise UnavailableModeException(photonic_modes, "Cannot add a configured circuit on non-photonic modes")
+
+        if modes[0] > 0:  # Barrier above detectors
+            ports = tuple(range(0, modes[0]))
+            self._components.append((ports, Barrier(len(ports), visible=True)))
+        for m in modes:
+            self._components.append(((m,), self._detectors[m]))
```
[the remark this answers is not in the document]
Comment (ericbrts) [reply perceval/components/abstract_processor.py:288]: We'll see, yes. But what kind of problem would you expect here?

**Review by Aubaert — APPROVED** [review]

File: tests/imgs/test_svg_dump_feed_forward_phys.svg
Comment (raksharuia) [inline tests/imgs/test_svg_dump_feed_forward_phys.svg:1]: Why do went the dotted line connecting D2 -> U(D2) to have the same color as classical modes?

File: tests/imgs/test_svg_processor_with_heralds_and_barriers_phys.svg
Comment (raksharuia) [inline tests/imgs/test_svg_processor_with_heralds_and_barriers_phys.svg:1]: Barrier is rendered overlapping with the text for the heralds. If shifting them is not too much of a nuisance, could we move them a bit?

File: perceval/rendering/circuit/phys_skin.py [hunk 2]
```diff
@@ -133,6 +145,30 @@ def get_shape(self, detector):
...
+        canvas.add_mline(origin + destination, stroke="white", stroke_width=PhysSkin._STROKE_WIDTH+2)
+        canvas.add_mline(origin + destination, **self.style[ModeType.CLASSICAL], stroke_dasharray="4,4")
+        origin[1] += offset_sign * 8
+        arrow_size = 3
+        for side in [-1, 1]:
+            canvas.add_mline(origin + [origin[0] + side * arrow_size, origin[1] - offset_sign*arrow_size],
+                             **self.style[ModeType.CLASSICAL])
+
+        # The actual component
+        canvas.add_rect((5, 10), 50 * w - 10, 50 * comp.m - 20, fill="lightgreen")
```
[the remark this answers is not in the document]
Comment (ericbrts) [reply perceval/rendering/circuit/phys_skin.py:169]: Yeah, you'd like an "intent" for colors. That's true it's used in a lot of places, we can think of it, but can it be in another story?

File: tests/imgs/test_svg_dump_feed_forward_phys.svg
Comment (ericbrts) [reply tests/imgs/test_svg_dump_feed_forward_phys.svg:1]: The actual physical link between a feed-forward controller (which is an electronics board) and a photonic chip is classical (an electrical wire).

I didn't check a template processor with heralds :/

File: tests/imgs/test_svg_processor_with_heralds_and_barriers_phys.svg
Comment (ericbrts) [reply tests/imgs/test_svg_processor_with_heralds_and_barriers_phys.svg:1]: Shifting them won't work when heralds are next to a closed box (like attached to a composite circuit with `recursive=False`.

**Pull request merged**